MenedzerBiblioteki/Books/BookRepository.cs

```diff
@@ -1,119 +0,0 @@
-﻿using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
-
-namespace MenedzerBiblioteki.Books
-{
-    class BookRepository : IBookRepository
```

Comment:
Generalnie to IBookRepository jest abstrakcją dostępu do danych. U Ciebie jest jeszcze IFileService, co już trochę mocno betonuje implementacje - mocno mówi o tym, że program musi czytać coś z pliku. Weź pod uwagę, że moglibyśmy też chcieć skorzystać np z bazy danych. Nejlepiej tworzyć oddzielne implementację IBookRepository pod mechanizm dostępu do danych - czyli jeden dla XMLa i jeden dla JSONa.